File: this-is-coding-test/11장_곱하기_혹은_더하기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String S = br.readLine();
        String[] splitWord = S.split("");

        int sums = 0;

        for (int i = 0; i < splitWord.length; i++) {
            int number = Integer.parseInt(splitWord[i]);

            if (sums < 2 || number < 2) {
                sums += number;
            } else {
                sums *= number;
            }
        }

        System.out.println(sums);
    }
}

File: this-is-coding-test/11장_곱하기_혹은_더하기.js
S = '02984';
const numbers = S.split('').map(v => parseInt(v));

let sums = numbers[0];

for (let i = 1; i < numbers.length; i++) {
    const number = numbers[i];

    // 더하기
    // 곱했을 때 0 일 경우, number 가 1일 경우
    if (sums * number === 0 || number === 1) {
        sums += number;
        continue;
    }

    sums *= number;
}

console.log(sums);
